src/Numerics/ODESolvers/AdditiveRungeKuttaMethod.jl

```diff
@@ -790,19 +833,115 @@ function ARK2GiraldoKellyConstantinescu(
         RT(1 / (2 * sqrt(2))) RT(1 / (2 * sqrt(2))) RT(1 - 1 / sqrt(2))
     ]
 
-    RKB = [RT(1 / (2 * sqrt(2))), RT(1 / (2 * sqrt(2))), RT(1 - 1 / sqrt(2))]
-    RKC = [RT(0), RT(2 - sqrt(2)), RT(1)]
+    RKB_explicit = [RT(1 / (2 * sqrt(2))), RT(1 / (2 * sqrt(2))), RT(1 - 1 / sqrt(2))]
+    RKC_explicit = [RT(0), RT(2 - sqrt(2)), RT(1)]
+    # For this ARK method, both RK methods share the same
+    # B and C vectors in the Butcher table
+    RKB_implicit = RKB_explicit
+    RKC_implicit = RKC_explicit
 
-    Nstages = length(RKB)
+    Nstages = length(RKB_explicit)
 
     AdditiveRungeKutta(
         F,
         L,
         backward_euler_solver,
         RKA_explicit,
         RKA_implicit,
-        RKB,
-        RKC,
+        RKB_explicit,
+        RKB_implicit,
+        RKC_explicit,
+        RKC_implicit,
+        split_explicit_implicit,
+        variant,
+        Q;
+        dt = dt,
+        t0 = t0,
+        nsubsteps = nsubsteps,
+    )
+end
+
+"""
+Trap2LockWoodWeller(F, L, backward_euler_solver, Q; dt, t0, nsubsteps,
+                    split_explicit_implicit, variant)
+
+This function returns an [`AdditiveRungeKutta`](@ref) time stepping object,
+    see the documentation of [`AdditiveRungeKutta`](@ref) for arguments definitions.
+    This time stepping object is intended to be passed to the `solve!` command.
+
+The time integrator scheme used is Trap2(2,3,2) with δ_s = 1, δ_f = 0, from
+the following reference
+
+### References
+    @article{Ascher1997,
+      title = {Numerical analyses of Runge–Kutta implicit–explicit schemes
+               for horizontally explicit, vertically implicit solutions of
+               atmospheric models},
+      author = {S.-J. Lock and N. Wood and H. Weller},
+      volume = {140},
+      number = {682},
+      pages = {1654-1669},
+      year = {2014},
+      journal = {Quarterly Journal of the Royal Meteorological Society},
+      publisher = {{RMetS}}
+    }
```

Comment:
Can we put this in `docs/bibliography/bib`?

Reply:
Oh yes! I followed what was done in the other functions above.

Reply:
Done in [71772a9](https://github.com/CliMA/ClimateMachine.jl/pull/1978/commits/71772a9907f033d5c89e474f70afcb8800378561)